subsys/bluetooth/mesh/Kconfig

```diff
@@ -94,67 +94,95 @@ endif # BT_MESH_RELAY
-	  conjunction with other NLC profiles, use CONFIG_BT_MESH_NLC_PERF_LIGHTNESS_CTRL instead.
+	  This option is deprecated. Use individual profile options instead.
+	  Enables performance configurations required for the Basic Scene Selector,
+	  Dimming Control, Energy Monitor, Occupancy Sensor, and Ambient Light
+	  Sensor NLC profiles.
+
+menu "NLC Profile performance configuration"
+
+config BT_MESH_NLC_PERF_SCENE_SELECTOR
+	bool "Basic Scene Selector NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_BASELINE
+
+config BT_MESH_NLC_PERF_DIMMING_CONTROL
+	bool "Dimming Control NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_BASELINE
+
+config BT_MESH_NLC_PERF_ENERGY_MONITOR
+	bool "Energy Monitor NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_BASELINE
+
+config BT_MESH_NLC_PERF_OCCUPANCY_SENSOR
+	bool "Occupancy Sensor NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_BASELINE
+
+config BT_MESH_NLC_PERF_AMBIENT_LIGHT_SENSOR
+	bool "Ambient Light Sensor NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_BASELINE
 
 config BT_MESH_NLC_PERF_LIGHTNESS_CTRL
-	bool "Enable strict performance configurations required for the Basic Lightness Controller"
-	help
-	  Enables required performance configurations for the
-	  NLC Basic Lightness Controller. The NLC Basic Lightness Controller
-	  share most performance configurations with the common NLC profiles
-	  (CONFIG_BT_MESH_NLC_PERF_DEFAULT) but with more strict requirements for a
-	  subset of the configurations.
+	bool "Basic Lightness Controller NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_STRICT
 
-endchoice
+config BT_MESH_NLC_PERF_HVAC_INTEGRATION
+	bool "HVAC Integration NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_STRICT
+
+endmenu
+
+endif # BT_MESH_NLC_PERF_CONF
 
 config BT_MESH_CRPL
-	default 32 if BT_MESH_NLC_PERF_DEFAULT
-	default 255 if BT_MESH_NLC_PERF_LIGHTNESS_CTRL
-	range 32 65535 if BT_MESH_NLC_PERF_DEFAULT
-	range 255 65535 if BT_MESH_NLC_PERF_LIGHTNESS_CTRL
+	default 255 if BT_MESH_NLC_PERF_STRICT
+	default 32 if BT_MESH_NLC_PERF_BASELINE
+	range 255 65535 if BT_MESH_NLC_PERF_STRICT
+	range 32 65535 if BT_MESH_NLC_PERF_BASELINE
 
 config BT_MESH_MODEL_GROUP_COUNT
-	default 32 if BT_MESH_NLC_PERF_LIGHTNESS_CTRL
-	range 32 4096 if BT_MESH_NLC_PERF_LIGHTNESS_CTRL
-
-if BT_MESH_NLC_PERF_CONF
+	default 32 if BT_MESH_NLC_PERF_STRICT
+	range 32 4096 if BT_MESH_NLC_PERF_STRICT
 
 config BT_MESH_MSG_CACHE_SIZE
-	default 64
-	range 64 65535
+	default 64 if BT_MESH_NLC_PERF_CONF
+	range 64 65535 if BT_MESH_NLC_PERF_CONF
 
 config BT_MESH_APP_KEY_COUNT
-	default 3
-	range 3 4096
+	default 3 if BT_MESH_NLC_PERF_CONF
+	range 3 4096 if BT_MESH_NLC_PERF_CONF
 
 config BT_MESH_SUBNET_COUNT
-	default 2
-	range 2 4096
+	default 2 if BT_MESH_NLC_PERF_CONF
+	range 2 4096 if BT_MESH_NLC_PERF_CONF
 
 config BT_MESH_PROXY_FILTER_SIZE
-	default 8
-	range 8 32767
-
-endif
```

Comment:
Hm... why did you move it up? Seems this was correct place, and you do not need adding dependency on `BT_MESH_NLC_PERF_CONF` to all parameters above. Did I miss anything?